src/core/batch/parquetSchema.ts

```diff
@@ -80,6 +109,55 @@ export const ReactionSchema = {
   signature: { type: "BYTE_ARRAY" },
 };
 
+/**
+ * ReactionBloomFilter: bloom filter options for batching reaction messages
+ */
 export const ReactionBloomFilterOptions = {
   bloomFilters: [{ column: "emoji" }, { column: "fromId" }, { column: "inReplyTo" }],
 };
+
+/**
+ * getSchemaFor() takes DSNPType and returns its corresponding parquet schema
+ *
+ * @param   dsnpType a dsnpType
+ * @returns The corresponding parquet schema
+ */
+export const getSchemaFor = (dsnpType: DSNPType): Schema => {
+  switch (dsnpType) {
+    case DSNPType.GraphChange:
+      return GraphChangeSchema;
+    case DSNPType.Broadcast:
+      return BroadcastSchema;
+    case DSNPType.Reply:
+      return ReplySchema;
+    case DSNPType.Reaction:
+      return ReactionSchema;
+    case DSNPType.Profile:
+      return ProfileSchema;
+    default:
+      throw new Error(`Invalid DSNP type: ${dsnpType}`);
+  }
```

Comment:
By removing the default, TypeScript will throw a build failure for incomplete, but it will not if there is a default. However as I think you were thinking, that doesn't give protection at run time. So I think we just need to move the throw to below the switch:

```suggestion
  }
  throw new Error(`Invalid DSNP type: ${dsnpType}`);
```